docs/development.rst

```diff
@@ -445,6 +448,86 @@ finally force-push to the remote topic branch on your fork::
+
+We also require that the demographic model is fully documented, including a unique
+identifier and descriptions and information to find to its original published source.
+The demographic model should include the following:
+
+* ``id``: A unique, short-hand identifier for this demographic model. This ``id``
+  contains a short description written in camel case, followed by an underscore, and then
+  four characters (the number of sampled populations, the first letter of the name of the
+  first author, and the year the study was published). For example, the Gutenkunst et al.
+  (2009) Out of Africa demographic model has the ``id`` "OutOfAfrica_3G09". 
+* ``description``: A brief one-line description of the demographic model.
+* ``long_description``: A longer description (say, a concise paragraph) that describes
+  the model in more detail.
+* ``populations``: A list of ``stdpopsim.Population`` objects, which have their own
+  ``id`` and ``description``. For example, the Thousand Genomes Project Yoruba panel
+  could be defined as ``stdpopsim.Population(id="YRI", description="1000 Genomes YRI (Yorubans)")``.
+* ``citations``: A list of ``stdpopsim.Citation`` objects for the appropriate citation
+  for this model. The citation object requires author, year, and doi information, and
+  a specified reason for citing this model.
+* ``generation_time``: The generation time for this species in years.
+
+If this is your first time implementing a demographic model in `stdpopsim`, it's a good
+idea to take some time browsing the Catalog and species' demographic models in the
+source code to see how existing models are typically written and documented. If you have
+any questions or confusion about formatting or implementing demographic models, please
+don't hesitate to open an `issue <http://github.com/popgensims/stdpopsim/issues>`_ --
+we're more than happy to answer any questions and help get you up and running.
+
+-----------------------------------
+What models are appropriate to add?
+-----------------------------------
+
+`Stdpopsim` supports any demographic model from the published literature that gives
+enough information to be able to define `msprime` demography objects. At a minimum, that
+includes population sizes and the timing of demographic events. These values need to
+either be given in "physical" units (that is, raw population sizes and time units in
+generations), or be able to be converted to physical units using, e.g., mutation rates
+used in the published study.
+
+Note that it is not necessary that the demographic model is attached to a particular
+species. `Stdpopsim` contains a collection of generic models that are widely used in
+developing and testing inference methods. If there is a generic model that does not
+currently exist in our catalog but would be useful to include, we also welcome those
+contributions. Again, there should also be a citation given for generic models.
```

Comment:
I feel that citations shouldn't be a hard requirement for generic models. But maybe we've already exhausted those generics for which no citation is needed? We currently have two or three generic models, depending on how you count them. There's `PiecewiseConstantSize`, `IsolationWithMigration`, and arguably `HomSap/Zigzag_1S14` is also a generic model. The first two don't have citations, but the third does.

Or am I wrong? Should we add citations for `PiecewiseConstantSize` and `IsolationWithMigration`?

Reply:
Seems reasonable to say we want a citation, but can make an exception if it turns out to be a super old and obvious one.

Reply:
Right.. I think it makes sense to at least make clear that a generic model should be one that is commonly tested against by methods developers, or something along those lines. I just wanted to avoid the case where someone spends time implementing their own somewhat niche toy model, just for us to reject that contribution because it isn't more widely used.